File: frameNet/src/main/java/org/sqlunet/framenet/sql/FnLayerQueryFromAnnoSetId.java
/*
 * Copyright (c) 2019. Bernard Bou <1313ou@gmail.com>.
 */

package org.sqlunet.framenet.sql;

import android.database.sqlite.SQLiteDatabase;

/**
 * FrameNet layer query from annoSet
 *
 * @author <a href="mailto:1313ou@gmail.com">Bernard Bou</a>
 */
class FnLayerQueryFromAnnoSetId extends FnLayerQuery
{
	/**
	 * <code>QUERY</code> is the SQL statement
	 */
	static private final String QUERY = SqLiteDialect.FrameNetLayerQueryFromAnnoSetId;

	/**
	 * Constructor
	 *
	 * @param connection      connection
	 * @param targetAnnoSetId target annoSet id
	 */
	public FnLayerQueryFromAnnoSetId(final SQLiteDatabase connection, final long targetAnnoSetId)
	{
		super(connection, targetAnnoSetId, FnLayerQueryFromAnnoSetId.QUERY);
	}
}

File: frameNet/src/main/java/org/sqlunet/framenet/sql/FnLayerQueryFromAnnoSetId.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.framenet.sql

import android.database.sqlite.SQLiteDatabase

/**
 * FrameNet layer query from annoSet
 *
 * @param connection      connection
 * @param targetAnnoSetId target annoSet id
 *
 * @author [Bernard Bou](mailto:1313ou@gmail.com)
 */
internal class FnLayerQueryFromAnnoSetId(connection: SQLiteDatabase, targetAnnoSetId: Long) : FnLayerQuery(connection, targetAnnoSetId, QUERY) {

    companion object {
        /**
         * `QUERY` is the SQL statement
         */
        private const val QUERY = SqLiteDialect.FrameNetLayerQueryFromAnnoSetId
    }
}
